src/ripper.rb

```diff
@@ -325,6 +325,30 @@ def on_comment(body)
       end
     end
   end
+
+  # Handles __END__ syntax, which allows individual scripts to keep content
+  # after the main ruby code that can be read through DATA.
+  module Ending
+    def initialize(source, *args)
+      super(source, *args)
+      @source = source
+      @ending = nil
+    end
+
+    def self.prepended(base)
+      base.attr_reader :source, :ending
+    end
+
+    def on___end__(body)
+      @ending = super(source.split("\n")[lineno..-1].join("\n"))
+    end
+
+    def on_program(*body)
+      super(*body).tap do |sexp|
+        sexp[:body][0][:body] << ending if ending
+      end
+    end
+  end
```

Comment:
Awesome! Good to know I was approaching this correctly - my PR was quite close to this 💯